packages/service-library/tests/test_redis_utils.py

```diff
@@ -52,9 +36,13 @@ def lock_name(faker: Faker) -> str:
 
 
 async def _contained_client(
-    redis_service: RedisSettings, lock_name: str, task_duration: float
+    get_redis_client_sdk: Callable[
+        [RedisDatabase], AbstractAsyncContextManager[RedisClientSDK]
+    ],
+    lock_name: str,
+    task_duration: float,
 ) -> None:
-    async with get_redis_client_sdk(redis_service) as redis_client_sdk:
+    async with get_redis_client_sdk(RedisDatabase.RESOURCES) as redis_client_sdk:
```

Comment:
same here

Reply:
I don't require a session and factory scope here .I required a callable to crate and destry the entire connection